sdk/clientcore/http-netty4/src/main/java/io/clientcore/http/netty4/NettyHttpClient.java

```diff
@@ -143,67 +150,123 @@ protected void initChannel(Channel ch) throws SSLException {
-                }
+        awaitLatch(latch);
+
+        ResponseStateInfo info = responseReference.get();
+        if (info == null) {
+            throw LOGGER.throwableAtError().log(errorReference.get(), CoreException::from);
+        }
 
-                Netty4ResponseHandler responseHandler
-                    = new Netty4ResponseHandler(request, responseReference, errorReference, latch);
-                channel.pipeline().addLast(responseHandler);
+        Response<BinaryData> response;
+        if (info.isChannelConsumptionComplete()) {
+            // The network response is already complete, handle creating our Response based on the request method and
+            // response headers.
+            BinaryData body = BinaryData.empty();
+            ByteArrayOutputStream eagerContent = info.getEagerContent();
+            if (info.getResponseBodyHandling() != ResponseBodyHandling.IGNORE && eagerContent.size() > 0) {
+                // Set the response body as the first HttpContent received if the request wasn't a HEAD request and
+                // there was body content.
+                body = BinaryData.fromBytes(eagerContent.toByteArray());
+            }
 
-                Throwable earlyError = errorReference.get();
-                if (earlyError != null) {
-                    // If an error occurred between the connect and the request being sent, don't proceed with sending
-                    // the request.
-                    latch.countDown();
-                    return;
+            response = new Response<>(request, info.getStatusCode(), info.getHeaders(), body);
+        } else {
+            // Otherwise we aren't finished, handle the remaining content according to the documentation in
+            // 'channelRead()'.
+            BinaryData body = BinaryData.empty();
+            ResponseBodyHandling bodyHandling = info.getResponseBodyHandling();
+            Channel channel = info.getResponseChannel();
+            if (bodyHandling == ResponseBodyHandling.IGNORE) {
+                // We're ignoring the response content.
+                CountDownLatch drainLatch = new CountDownLatch(1);
+                channel.pipeline().addLast(new Netty4EagerConsumeChannelHandler(drainLatch, ignored -> {
+                }));
+                channel.config().setAutoRead(true);
+                awaitLatch(drainLatch);
+            } else if (bodyHandling == ResponseBodyHandling.STREAM) {
+                // Body streaming uses a special BinaryData that tracks the firstContent read and the Channel it came
+                // from so it can be consumed when the BinaryData is being used.
+                // autoRead should have been disabled already but lets make sure that it is.
+                channel.config().setAutoRead(false);
+                String contentLength = info.getHeaders().getValue(HttpHeaderName.CONTENT_LENGTH);
```

Comment:
nit, don't we have a core util for this?

Reply:
Might be confusing this functionality with `CoreUtils.extractSizeFromContentRange`, but we could certainly look at adding a common utility method for parsing Content-Length